packages/near-api-js/src/transaction.ts

```diff
@@ -97,6 +97,26 @@ export function deleteAccount(beneficiaryId: string): Action {
 export class Signature extends Assignable {
     keyType: KeyType;
     data: Uint8Array;
+    
+    static borshDeserialize(reader: BinaryReader) {
+        const keyType = reader.readU8();
+        let data: Uint8Array;
+        switch (keyType) {
+            case KeyType.ED25519: 
+                data = reader.readFixedArray(64);
+                break;
+            case KeyType.SECP256K1: 
+                data = reader.readFixedArray(65);
+                break;
+            default: throw new Error(`Unknown key type ${keyType}`);
+        }
+        return new Signature({keyType, data});
+    }
+    
+    borshSerialize(writer: BinaryWriter) {
```

Comment:
Should we not create a new empty `BinaryWriter` here? If there's a use case for appending this to an existing buffer then we should create an empty `BinaryWriter` when none is provided.

Reply:
Answered here: https://github.com/near/near-api-js/pull/985/files#r981167051